pkg/util/pods.go

```diff
@@ -376,14 +425,20 @@ func checkPending(pod corev1.Pod, timeout time.Duration, now time.Time) error {
 			panic(fmt.Sprintf("invalid container status: %#v", s))
 		}
 	}
+	// One of the containers must be in the "waiting" state, by exclusion.
```

Comment:
[Apparently not](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/3281/pull-ci-openshift-ci-tools-master-e2e/1625886302459138048#1:build-log.txt%3A52).